aiden-recommender/aiden_recommender/models.py

```diff
@@ -1,7 +1,118 @@
+from __future__ import annotations
 from datetime import datetime
-from typing import Optional
+import json
+from typing import AsyncGenerator, Callable, Optional, Any
 
-from pydantic import AliasChoices, BaseModel, Field
+from pydantic import BaseModel
+from abc import ABC, abstractmethod
+from mistralai.models.embeddings import EmbeddingObject
+from qdrant_client.models import PointStruct
+from aiden_recommender.constants import JOB_COLLECTION
+from aiden_recommender.tools import async_zyte_client, async_redis_client, async_mistral_client, async_qdrant_client
+import hashlib
+import uuid
+
+
+def reference_to_uuid(reference: str) -> uuid.UUID:
+    return uuid.UUID(hashlib.md5(reference.encode()).hexdigest()).hex
```

Comment:
func is returning a string 
```suggestion
def reference_to_uuid(reference: str) -> uuid.UUID:
    return uuid.UUID(hashlib.md5(reference.encode()).hexdigest())
```